### ISSUE TITLE
strange blunder

### ISSUE THREAD

**Issue by tissatussa** [mkd/gargantua#2]
In this position Gargantua v1.0 NNUE made a huge blunder, which is impossible to explain : 30. Qg7+ ??  I guess the NN logic made a huge mistake here.

FEN: rq2rnk1/1p1bbp1p/1Pp1p2Q/p1P1P3/P1BP4/5Np1/5P2/2BRR1K1 w - - 0 30

![gargantua-blunder](https://github.com/mkd/gargantua/assets/1109281/c052402c-67ee-488b-8ec2-c35c12ad952c)

[ see [https://lichess.org/hdmzsNvY](https://lichess.org/hdmzsNvY) ]
The game was played against Mizar v3.0 in CuteChess, using 128 Mb Hash .. it was a 15 minute game .. according to the data (see PGN below) the engine took 21 seconds for this move and reached depth 18 (if i understand the CuteChess data correctly) .. but when i feed the concerning FEN into CuteChess and let Gargantua evaluate this position again, giving infinite time, the move Qg7+ isn't showed at any depth, and the engine finds best move Kg2 .. btw. it doesn't reach depth 18 : highest reached depth is 17, even after many minutes ..

[Event "engine vs engine"]
[Site "Holland"]
[Date "2024.01.12"]
[Round "?"]
[White "Gargantua v1.0 NNUE"]
[Black "Mizar v3.0"]
[Result "0-1"]
[ECO "E11"]
[GameDuration "00:35:50"]
[Opening "Bogo-Indian defense, Gruenfeld variation"]
[PlyCount "94"]
[TimeControl "900+10"]

1. d4 {0.00/16 40s} Nf6 {-0.03/10 30s} 2. c4 {0.00/16 39s} e6 {+0.29/10 30s} 3. Nf3 {0.00/15 38s} Bb4+ {+0.39/9 29s} 4. Nbd2 {+0.72/13 37s} Nc6 {+0.12/10 28s} 5. a3 {+1.25/15 36s} Be7 {+0.15/10 28s} 6. e3 {+1.36/16 36s} O-O {+0.11/10 27s} 7. b4 {+1.65/16 35s} d5 {+0.20/10 27s} 8. Qc2 {+1.53/14 34s} a5 {+0.62/9 26s} 9. b5 {+2.38/17 33s} Na7 {+0.69/9 26s} 10. c5 {+3.38/16 32s} Bd7 {+0.63/9 25s} 11. a4 {+3.34/18 32s} Ne4 {-0.09/9 24s} 12. Nxe4 {+6.07/19 31s} dxe4 {+0.46/8 24s} 13. Qxe4 {+6.39/18 30s} c6 {-0.32/8 24s} 14. b6 {+6.42/17 29s} Nc8 {-1.40/10 23s} 15. Bc4 {+6.66/17 29s} Bf6 {-1.49/9 23s} 16. O-O {+7.19/17 28s} Be8 {-1.56/9 22s} 17. Ba3 {+7.91/17 28s} Ne7 {-1.31/9 22s} 18. Qc2 {+8.01/16 27s} Nd5 {-1.27/8 21s} 19. e4 {+8.64/17 26s} Nf4 {-1.31/9 21s} 20. Rad1 {+8.75/17 26s} Ng6 {-1.35/9 21s} 21. g3 {+8.69/17 25s} Bd7 {-1.37/9 20s} 22. e5 {+9.39/17 25s} Be7 {-1.37/10 20s} 23. Rfe1 {+9.10/18 24s} Qb8 {-1.33/9 20s} 24. h4 {+9.57/18 24s} Re8 {-1.47/9 19s} 25. h5 {+10.84/19 23s} Nf8 {-1.71/10 19s} 26. h6 {+11.37/19 23s} gxh6 {-1.85/9 19s} 27. Bc1 {+12.89/18 23s} h5 {-1.66/10 18s} 28. Qd2 {+12.08/18 22s} h4 {-2.30/9 18s} 29. Qh6 {+13.63/18 22s} hxg3 {-1.95/9 18s} 30. Qg7+ {0.00/18 21s} Kxg7 {+9.36/10 18s} 31. Ng5 {-10.58/17 21s} f6 {+10.96/9 17s} 32. Ne4 {-12.79/18 21s} gxf2+ {+11.59/10 17s} 33. Kxf2 {-9.08/16 20s} fxe5 {+12.04/9 17s} 34. Rh1 {-8.78/16 20s} Kh8 {+12.85/8 17s} 35. Rdg1 {-7.99/16 20s} exd4 {+12.82/9 16s} 36. Nd6 {-8.35/16 19s} Bxd6 {+14.96/10 16s} 37. cxd6 {0.00/17 19s} Qxd6 {+14.96/8 16s} 38. Ke2 {0.00/17 19s} Qb4 {+16.19/8 16s} 39. Bd3 {0.00/17 18s} e5 {+16.12/8 16s} 40. Bg5 {-17.13/16 18s} Re6 {+15.97/8 15s} 41. Kf2 {0.00/16 18s} Qb2+ {+17.62/8 15s} 42. Ke1 {0.00/15 18s} Qc3+ {+19.08/9 15s} 43. Ke2 {-26.20/14 17s} e4 {+20.38/8 15s} 44. Ba6 {0.00/15 17s} Qc2+ {+25.71/8 15s} 45. Ke1 {-28.62/13 17s} e3 {+29.28/8 15s} 46. Rg2 {-29.60/16 17s} Qxg2 {+32.80/8 14s} 47. Kd1 {-M12/24 16s} Qd2# {+M1/7 14s, Black mates} 0-1

**Comment by mkd** [comment]
Hi Roelof,

thanks for reaching out. I don't have CuteChess on my Mac, I tried ScidvsPC
as well as the command line, and it always finds Kg2 as the best move after
reaching depth 18. It never shows Qg7+.

Will try to reproduce this with a CuteChess at work or somewhere else, when
I have the chance.


Regards,


--
Claudio M. Camacho
Global Brand & Marketing Executive | Board Director | Advisor

www.claudiocamacho.com

twitter: twitter.com/claudiomkd
linkedin: http://www.linkedin.com/in/claudiocamacho
medium: https://claudiocamacho.medium.com/

**Comment by tissatussa** [comment]
@mkd i don't think this has anything to do with CuteChess .. often it's not possible to reproduce such error (!?) .. i guess it has something to do with the hash management in combination with the NN logic -- that's just my feeling.
